src/FluentSchema.js

```diff
@@ -627,459 +71,33 @@ const FluentSchema = (
-        )
-      return {
-        ...memo,
-        [prop]: Array.isArray(schema)
-          ? schema
-          : omit(schema.valueOf(), ['$schema', 'type', 'definitions']),
-      }
-    }, {})
-    return setAttribute({ schema, ...options }, [
-      'dependencies',
-      values,
-      'object',
-    ])
-  },
-
-  /**
-   * If the instance is an object, this keyword validates if every property name in the instance validates against the provided schema.
-   * Note the property name that the schema is testing will always be a string.
-   *
-   * @param {FluentSchema} value
-   * {@link reference|https://json-schema.org/latest/json-schema-validation.html#rfc.section.6.5.7}
-   * @returns {FluentSchema}
-   */
-
-  propertyNames: value => {
-    if (!isFluentSchema(value))
-      throw new Error("'propertyNames' must be a FluentSchema")
-    return setAttribute({ schema, ...value.valueOf() }, [
-      'propertyNames',
-      omit(value.valueOf(), ['$schema']),
-      'object',
-    ])
-  },
-
-  /**
-   * Set a property to type null
-   *
-   * {@link reference|https://json-schema.org/latest/json-schema-validation.html#rfc.section.6.1.1}
-   * @returns {FluentSchema}
-   */
-
-  asNull: () => FluentSchema({ schema: { ...schema }, options }).as('null'),
-
-  /**
-   * @private set a property to a type. Use asString asNumber etc.
-   * @returns {FluentSchema}
-   */
-  as: type => {
-    return setAttribute({ schema, ...options }, ['type', type])
-  },
-
-  /**
-   * This validation outcome of this keyword's subschema has no direct effect on the overall validation result.
-   * Rather, it controls which of the "then" or "else" keywords are evaluated.
-   * When "if" is present, and the instance successfully validates against its subschema, then
-   * validation succeeds against this keyword if the instance also successfully validates against this keyword's subschema.
-   *
-   * @param {FluentSchema} ifClause
-   * @param {FluentSchema} thenClause
-   * {@link reference|https://json-schema.org/latest/json-schema-validation.html#rfc.section.6.6.1}
-   * @returns {FluentSchema}
-   */
-
-  ifThen: (ifClause, thenClause) => {
-    if (!isFluentSchema(ifClause))
-      throw new Error("'ifClause' must be a FluentSchema")
-    if (!isFluentSchema(thenClause))
-      throw new Error("'thenClause' must be a FluentSchema")
-
-    const ifClauseSchema = omit(ifClause.valueOf(), [
-      '$schema',
-      'definitions',
-      'type',
-    ])
-    const thenClauseSchema = omit(thenClause.valueOf(), [
-      '$schema',
-      'definitions',
-      'type',
-    ])
-
-    return FluentSchema({
-      schema: {
-        ...schema,
-        if: patchIdsWithParentId({
-          schema: ifClauseSchema,
-          ...options,
-          parentId: '#if',
-        }),
-        then: patchIdsWithParentId({
-          schema: thenClauseSchema,
-          ...options,
-          parentId: '#then',
-        }),
-      },
+  asArray: () =>
```

Comment:
I think we could just do `array()`, `string()`, `number()`, `object()`, etc. I think the `'as'` as lost any meaning with this refactoring.

Reply:
But it would break the API. I could create the new syntax and deprecate the old one.